LPS-200228 LPS-200227 LPS-200225

lfbesada opened the pull request.

liferay-continuous-integration commented:
To conserve resources, the PR Tester does not automatically run for every pull.

If your code changes were already tested in another pull, reference that pull in this pull so the test results can be analyzed.

If your pull was never tested, comment &quot;ci:test&quot; to run the PR Tester for this pull.

sandrodw3 commented:
Just started reviewing :) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

sandrodw3 commented:
Pull request submitted to https://github.com/jkappler/liferay-portal/pull/257. [See changes here.](https://github.com/sandrodw3/liferay-portal/compare/14138a17af74~1...14138a17af74) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

lfbesada committed:
LPS-200227 Throws a LockedLayoutException if a user tries to change the classNameId or the classTypeId of a locked page

lfbesada committed:
LPS-200227 SF use the LayoutLockManager method

lfbesada committed:
LPS-200227 Handles LockedLayoutException by redirecting to the locked page view

lfbesada committed:
LPS-200227 Adds new getLock signature to specify the layout and the user ID instead of get them from theme display

lfbesada committed:
LPS-200227 Prevents NPE if layout is null

lfbesada committed:
LPS-200227 Takes the layout lock for this user before to be allowed to change the content type and release it after the action

lfbesada committed:
LPS-200227 Throws an UnsupportedOperationException if trying to change the content type and the feature flag is disabled

lfbesada committed:
LPS-200227 Adds back_url and back_url_title parameters

lfbesada committed:
LPS-200227 SF pass changeContentTypeURL as dropdownItem data instead of the additional properties

lfbesada committed:
LPS-200227 Disables the change content type option when the draft layout is locked by other user

lfbesada committed:
LPS-200228 SF extracts to reuse

lfbesada committed:
LPS-200228 Disable the change content type for the display pages that are being used

lfbesada committed:
LPS-200228 Adds the language key

lfbesada committed:
LPS-200228 BuildLang

lfbesada committed:
LPS-200228 Adds a ModelListener for LayoutPageTemplateEntry to prevent the content type changes when the display page is used by at least one asset.

I am using a model listener for this because trying to check it in the service layer will cause a circular dependency between AssetDisplayPageEntryLocalService and LayoutPageEntryLocalService. Checking it only in the action command would leave open the possibility to change the content type of a used display page by invoking directly the service method.

lfbesada committed:
LPS-200228 Handles properly the expected model listener exception

lfbesada committed:
LPS-200225 SF remove redundant condition as it is going to be always true

lfbesada committed:
LPS-200225 Hide all the behavior under the feature flag

lfbesada committed:
LPS-200225 Pass existsMappedContentType to DisplayPageActionDropdownItemsProvider and use it to prevent access directly to the edit action if the display page is mapped to a non-existing type or a subtype.

To achieve it, display the change content type popup, and redirect to the edition once the content type is fixed. So we are setting as action the changeContentType one but following the redirect to the edit page.

lfbesada committed:
LPS-200227 Baseline

lfbesada committed:
LPS-200225 SF cache in a variable to prevent calculating for each display page template

lfbesada committed:
LPS-200225 Adds a method to determine if the display page's mapped content type exists

lfbesada committed:
LPS-200225 Improve performance and simplify by caching the type and subtypes IDs using a map since there could be many display page templates
